packages/event-handler/src/rest/RouteHandlerRegistry.ts

```diff
@@ -22,29 +86,118 @@ class RouteHandlerRegistry {
+      if (this.#dynamicRoutesIndexMap.has(route.id)) {
+        this.#logger.warn(
+          `Handler for method: ${route.method} and path: ${route.path} already exists. The previous handler will be replaced.`
+        );
+        const index = this.#dynamicRoutesIndexMap.get(route.id)!;
+        this.#dynamicRoutes.splice(index, 1);
+      }
+
+      const index = this.#addDynamicRoute({
+        ...route,
+        ...compiled,
+      });
+
+      this.#dynamicRoutesIndexMap.set(route.id, index);
+    } else {
+      if (this.#staticRoutes.has(route.id)) {
+        this.#logger.warn(
+          `Handler for method: ${route.method} and path: ${route.path} already exists. The previous handler will be replaced.`
+        );
+      }
+      this.#staticRoutes.set(route.id, route);
+    }
 
     const routesByMethod = this.#routesByMethod.get(route.method) ?? [];
     routesByMethod.push(route);
     this.#routesByMethod.set(route.method, routesByMethod);
   }
 
+  /**
+   * Resolves a route handler for the given HTTP method and path.
+   *
+   * Static routes are checked first for exact matches. Dynamic routes are then
+   * checked in order of specificity (fewer parameters and more segments first).
+   * If no handler is found, it returns `null`.
+   *
+   * Examples of specificity (given registered routes `/users/:id` and `/users/:id/posts/:postId`):
+   * - For path `'/users/123/posts/456'`:
+   *   - `/users/:id` matches but has fewer segments (2 vs 4)
+   *   - `/users/:id/posts/:postId` matches and is more specific -> **selected**
+   * - For path `'/users/123'`:
+   *   - `/users/:id` matches exactly -> **selected**
+   *   - `/users/:id/posts/:postId` doesn't match (too many segments)
+   *
+   * @param method - The HTTP method to match
+   * @param path - The path to match
+   * @returns Route handler options or null if no match found
+   */
+  public resolve(method: HttpMethod, path: Path): RouteHandlerOptions | null {
+    const routeId = `${method}:${path}`;
+
+    const staticRoute = this.#staticRoutes.get(routeId);
+    if (staticRoute != null) {
+      return {
+        handler: staticRoute.handler,
+        rawParams: {},
+        params: {},
+      };
+    }
+
+    for (const route of this.#dynamicRoutes) {
+      if (route.method !== method) continue;
+
+      const match = route.regex.exec(path);
+      if (match?.groups) {
+        const params = match.groups;
+
+        const processedParams = processParams(params);
+
+        const validation = validateParams(processedParams);
+
+        if (!validation.isValid) {
+          throw new ParameterValidationError(validation.issues);
+        }
+
+        return {
+          handler: route.handler,
+          params: processedParams,
+          rawParams: params,
+        };
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Returns the total number of registered routes.
+   * @returns Total count of registered routes
+   */
   public getRouteCount(): number {
```

Comment:
I think we can probably get rid all three of these methods, I don't think we need them.

Reply:
Feel free to remove it, can't think of any use for this - and if we need it we can always add it back.